packages/@lwc/shared/src/api-version.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { isNumber } from './language';
+
+export const enum APIVersion {
+    V58_244_SUMMER_23 = 58,
+    V59_246_WINTER_24 = 59,
+}
+
+// These must be updated when the enum is updated.
+// It's a bit annoying to do have to do this manually, but this makes the file tree-shakeable,
+// passing the `verify-treeshakeable.js` test.
```

Comment:
This doesn't seem to bad to me. Just add it to the list of tasks just prior to feature freeze.

Reply:
I considered an alternative, which is to use the `/*#__PURE__*/` annotation. The problem with that is that the bundle size becomes larger. But it may be worth it if we can avoid a hard-to-remember manual task.

Also, we don't actually have to do this every feature freeze – only when we add a new `APIFeature`. If we skip a release, and there are no API changes in that release, then we can keep using the older one as the "latest".